### PR TITLE
fixed move semantics for sf::Cursor and added test for it.

### PR DESCRIPTION
**Pull request by alirezafour**
## Description

because of the forward declaration of `sf::priv::CursorImpl` on header file implementation of move constructor and move assignment operator on header file cause the code not to compile because `std::unique_ptr` need the implementation of `sf::priv::CursorImpl` to be able to delete it properly.

I moved the implementation into `Cursor.cpp` and added test for it.

This PR is related to the issue #2483

----
* [ ] Has this change been discussed on [the forum](https://en.sfml-dev.org/forums/index.php#c3) or in an issue before?
* [x] Does the code follow the SFML [Code Style Guide](https://www.sfml-dev.org/style.php)?
* [x] Have you provided some example/test code for your changes?
* [ ] If you have additional steps which need to be performed list them as tasks!

----
## Tasks

* [x] Tested on Linux
* [x] Tested on Windows
* [ ] Tested on macOS
* [ ] Tested on iOS
* [ ] Tested on Android

## How to test this PR?

```cpp
#include <SFML/Graphics.hpp>
#include <SFML/Window/Cursor.hpp>

int main()
{
    sf::RenderWindow window(sf::VideoMode({1280, 720}), "Minimal, complete and verifiable example");
    window.setFramerateLimit(60);
    sf::Cursor cursor(std::move(sf::Cursor()));


	if (cursor.loadFromSystem(sf::Cursor::Hand))
	     window.setMouseCursor(cursor);


    while (window.isOpen())
    {
        for (sf::Event event; window.pollEvent(event);)
        {
            if (event.type == sf::Event::Closed)
                window.close();
        }

        window.clear();
        window.display();
    }
}
```

### CODE REVIEW

**Comment by eXpl0it3r** [comment]
SFMLBuildMaster: Build this please.

**Comment by ChrisThrasher** [comment]
Thank you for fixing this! I think I see what's going on now. I assumed that 

```cpp
static_assert(std::is_nothrow_move_constructible_v<sf::Cursor>);
static_assert(std::is_nothrow_move_assignable_v<sf::Cursor>);
```

was enough to verify that moving a Cursor would compile but I now see that that we also need to compile code that moves a given type to be sure that moving that type will actually compiler.

I see a few increasingly cumbersome steps for verifying correct move semantics. Not all types need to take all steps but some types need more thorough testing than others.

1. Assert the correct type traits for move semantics (we already do this for all types) (good enough for types following the Rule of Zero)
1. Compile tests that perform move construction and move assignment (good enough for types whose internals allow for `= default`'ed move operations) (this is often done by our examples which use large swaths of the API but not all of the API)
1. Run said tests to confirm nothing crashes (this isn't possible for types like `sf::Cursor` for which we can't run the tests thanks to incomplete DRM support)
1. Assert some post condition to confirm the move worked as intended (this is difficult when many types are too opaque to inspect the internals to confirm the move operation succeeded)
1. Perform non-automated tested with a sample project to confirm the move operations are correct (we have done this before but it's very expensive)

File: test/Window/Cursor.test.cpp [hunk 1]
```diff
@@ -19,6 +19,19 @@ TEST_CASE("[Window] sf::Cursor" * doctest::skip(true))
...
+        const sf::Cursor movedCursor(std::move(cursor));
+    }
+
+    SUBCASE("Move assigment operator")
+    {
+        sf::Cursor cursor;
+        sf::Cursor movedCursor;
+        movedCursor = std::move(cursor);
+    }
+
```
Comment (ChrisThrasher) [inline test/Window/Cursor.test.cpp:34]:
```suggestion
    SUBCASE("Move semantics")
    {
        SUBCASE("Construction")
        {
            sf::Cursor       movedCursor;
            const sf::Cursor cursor(std::move(movedCursor));
        }

        SUBCASE("Assignment")
        {
            sf::Cursor movedCursor;
            sf::Cursor cursor;
            cursor = std::move(movedCursor);
        }
    }

```

**Review by ChrisThrasher — CHANGES_REQUESTED** [review]
Other than a minor style issue this is good to go

**Comment by ChrisThrasher** [comment]
SFMLBuildMaster: Build this please.

**Review by ChrisThrasher — APPROVED** [review]
Thank you!

**Comment by vittorioromeo** [comment]
@ChrisThrasher: this might be of interest: https://stackoverflow.com/questions/51901837/how-to-get-if-a-type-is-truly-move-constructible/51912859#51912859

**Review by eXpl0it3r — APPROVED** [review]

**Comment by eXpl0it3r** [comment]
Thanks! 🙂

**Pull request merged**